lib/plugins/aws/package/compile/functions.js

```diff
@@ -219,9 +218,20 @@ class AwsCompileFunctions {
         ? this.serverless.service.package.deploymentBucket
         : { Ref: 'ServerlessDeploymentBucket' };
 
-      functionResource.Properties.Code.S3Key = `${
-        this.serverless.service.package.artifactDirectoryName
-      }/${artifactFilePath.split(path.sep).pop()}`;
+      const serverlessPackagedArtifact = artifactFilePath.split(path.sep).indexOf('.serverless');
```

Comment:
I'm wondering if we could replace the logic that detects if there's `serverless` in path with something like:

```
artifactFilePath.split('.serverless').pop()
```

What do you think?

Reply:
As far as I can see it breaks tests on Windows, while this one (after applying suggestion mentioned in https://github.com/serverless/serverless/pull/8891#discussion_r574462870) works correctly resulting in the same S3Key format on *nix and Windows.

Reply:
Hello :wave: I've checked the previous commits and I see that you've using split on `.serverless/` which will indeed break as it includes a `/` - I think that using `.serverless` instead of `.serverless/` when splitting should work correctly here

Reply:
I'm afraid it'll produce S3Key like `some/path//path/to/artifact/artifact.zip` or `some/path/\path/to/artifact/artifact.zip` (depending on the OS). 

```js
artifactFilePath.split(`.serverless${path.sep}`).pop()
```

is what I used for now.